cmake/cjson-patch.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+# This PATCH_COMMAND script performs the following:
+# Replaces the following line in the cJSON CMakeLists.txt:
+# `cmake_minimum_required(VERSION 3.0)`
+# With the line:
+# `cmake_minimum_required(VERSION 3.24)`
+# This allows us to build it with CMake v4
+if [ "$(uname)" = 'Darwin' ]; then
+  # Unix sed
+  sed -e '2s/0/24/' -i '' CMakeLists.txt
+else
+  # GNU sed
+  sed -e '2s/0/24/' -i CMakeLists.txt
```

Comment:
This reads replace the `0` on line 2 with a `24`, and do it in place without a backup file.

Reply:
Since our version is pinned, line 2 is: `cmake_minimum_required(VERSION 3.0)` and becomes `cmake_minimum_required(VERSION 3.24)`